lib/credentials_from_file.c

```diff
@@ -41,46 +41,74 @@ bool write_normal_device_credentials(const char *dev_cred_file,
 		return false;
 	}
 #ifndef NO_PERSISTENT_STORAGE
-	sdow_t sdowriter, *sdow = &sdowriter;
 
-	if (!sdow_init(sdow)) {
+	sdow_t *sdow = sdo_alloc(sizeof(sdow_t));
```

Comment:
There is a possibility for sdo_alloc to return NULL. Add return code check.

Reply:
Done.